text/0000-set-helper.md

```diff
@@ -0,0 +1,106 @@
+- Start Date: 2020-02-20
+- Relevant Team(s): Ember.js
+- RFC PR: (after opening the RFC PR, update this with a link to it and update the file name)
+- Tracking: (leave this empty)
+
+# `set` helper
+
+## Summary
+
+This RFC introduces a new `set` helper like the one provided by
+[`ember-simple-set-helper`](https://github.com/pzuraq/ember-simple-set-helper).
+
+The `set` helper will always return a function that will change a value.
+
+```hbs
+<SelectCountry @update={{set this.country}} />
+```
+
+The new `set` helper can be used instead of the old `{{fn (mut ..)}}` combination.
+However this RFC does not deprecate `{{mut}}`.
```

Comment:
Why *not* deprecate `{{mut}}`, preferably with a timeline where there are one or more minor releases between the time when `{{set}}` is added and `{{mut}}` is deprecated so that addons can migrate without causing unactionable noise for consumers.

If there's a reason we *cannot* or *should not* deprecate `{{mut}}`, I think (a) we should spell that out here, but also (b) it undercuts the motivation for this RFC. We end up in a spot where there are two ways of doing something, one of them likely preferred/recommended but the other still supported and without a clear indication that you should move away from it to the new preferred/recommended approach. If there are gaps in functionality from `{{mut}}` which we believe need to be covered, it's fine to deliver them separately, but that should be expressly called out in the text of this RFC.

Reply:
Honestly I think there are *maybe* gaps, be we need to find them. We have no replacement for the *mutable attribute* syntax. I think long term we should deprecate `mut`, but this will be a first step in that direction. A future RFC should deprecate mut, and maybe additional RFCs are required for the gaps. I will try to make this clear in the RFC. Thanks for pointing it out.